### PR TITLE
[pigeon] More integration tests

### PR DESCRIPTION
**Pull request by tarrinneal**
Adds a large number of async integration tests for all platforms.

Also adds List, Map, nullable List and Map, Null list and map, and async of all of those as well.

Some of this work may seem repetitive/unnecessary, but I think it's best to cover as many angles as possible and most of this work was copy/paste.

Continued work towards https://github.com/flutter/flutter/issues/116374.

### CODE REVIEW

File: packages/pigeon/platform_tests/test_plugin/windows/test_plugin.h [hunk 1]
```diff
@@ -178,7 +235,3 @@ class TestPlugin : public flutter::Plugin,
   std::unique_ptr<core_tests_pigeontest::FlutterIntegrationCoreApi>
       flutter_api_;
 };
-
-}  // namespace test_plugin
-
-#endif  // FLUTTER_PLUGIN_TEST_PLUGIN_H_
```
Comment (stuartmorgan) [inline packages/pigeon/platform_tests/test_plugin/windows/test_plugin.h:184]: This won't compile, which is probably why CI is red.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/pigeon/pigeons/core_tests.dart [hunk 2]
```diff
@@ -194,12 +194,42 @@ abstract class HostIntegrationCoreApi {
...
   @ObjCSelector('echoAsyncString:')
   @SwiftFunction('echoAsync(_:)')
   String echoAsyncString(String aString);
 
+  /// Returns the passed in Uint8List asynchronously.
+  @async
+  @ObjCSelector('echoAsyncUint8List:')
+  @SwiftFunction('echoAsync(_:)')
+  Uint8List echoAsyncUint8List(Uint8List aUint8List);
+
```
[the remark this answers is not in the document]
Comment (tarrinneal) [reply packages/pigeon/pigeons/core_tests.dart:246]: Way ahead of you

File: packages/pigeon/platform_tests/test_plugin/windows/test_plugin.h [hunk 3]
```diff
@@ -1,6 +1,3 @@
-// Copyright 2013 The Flutter Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
```
[the remark this answers is not in the document]
Comment (tarrinneal) [reply packages/pigeon/platform_tests/test_plugin/windows/test_plugin.h:3]: Didn't even realize I had. I'll fix this

**Review by stuartmorgan — APPROVED** [review]
LGTM! I pushed a fix for Windows.

Unfortunately the usability of nullable async returns in the C++ generator is worse than I expected; it's not doing implicit conversion chaining from the raw value to the `ErrorOr<std::optional<WhateverType>>` for some reason. I'm wondering if it's a compiler bug since the exact same implicit conversion works for returning values.

**Pull request merged**